packages/beacon-node/src/api/impl/beacon/state/index.ts

```diff
@@ -59,12 +59,12 @@ export function getBeaconStateApi({
       const {state, executionOptimistic} = await resolveStateId(chain, stateId);
       const currentEpoch = getCurrentEpoch(state);
       const {validators, balances} = state; // Get the validators sub tree once for all the loop
-      const {pubkey2index} = chain.getHeadState().epochCtx;
+      const pubkey2indexFn = chain.getHeadState().epochCtx.getValidatorIndex;
```

Comment:
This is dangerous to do in javascript since you are referencing a class method without binding it to the class this. This code should not work if getValidatorindex body access the `this` variable. Even then I would prefer to not encourage this pattern within lodestar. Just pass the epochCtx variable around unless there is a really good reason of why not to do that.